File: 백준/Bronze/9506. 약수들의 합/약수들의 합.py
while True:
    li=[]
    n=int(input())
    
    if n==-1:
        break
        
    for i in range(1,n):
        if n%i==0:
            li.append(i)
    
    if sum(li)==n:
        print(n, "= ", end='')
        print(*li, sep=' + ')
    else:
        print(n,"is NOT perfect.")

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.java
import java.util.*;
import java.io.*;

class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        while(true){
            ArrayList<Integer> arr = new ArrayList<>();
            int n = Integer.parseInt(br.readLine());
            if(n==-1) break;
            
            for(int i = 1; i<n; i++){
                if(n%i==0) arr.add(i);
            }
            int sum = 0;
            for(int i:arr) sum += i;
            if(sum==n){
                sb.append(n).append(" = ");
                for(int i = 0; i<arr.size()-1; i++) sb.append(arr.get(i)).append(" + ");
                sb.append(arr.get(arr.size()-1));
            }
            else{
                sb.append(n).append(" is NOT perfect.");
            }
            sb.append("\n");
        }
        
        System.out.println(sb);
    }
}